File: PROGRAMMING FOUNDATION - JAVA/model/Book.java
package model;

class Book {
    private String bookId;
    private String title;
    private String author;
    private boolean isAvailable;

    public Book(String bookId, String title, String author) {
        this.bookId = bookId;
        this.title = title;
        this.author = author;
        this.isAvailable = false; // false = available
    }

    public String getBookId() {
        return bookId;
    }

    public String getBookTitle() {
        return title;
    }

    public boolean isAvailable() {
        return isAvailable;
    }

    public void markAvailable(boolean status) {
        this.isAvailable = status;
    }

    public void printDetails() {
        System.out.println("BookId: " + bookId + " | Title: " + title + ", Writer: " + author +
                (isAvailable ? " (Issued)" : " (Available)"));
    }
}

File: PROGRAMMING FOUNDATION - PYTHON/LIBRARY_MANAGEMENT_SYSTEM/Book.py
class Book:
    def __init__(self, title, author, isbn):
        self._title = title
        self._author = author
        self._isbn = isbn
        self._availability = True

    def get_title(self):
        return self._title
    
    def get_author(self):
        return self._author

    def get_isbn(self):
        return self._isbn

    def is_available(self):
        return self._availability
    
    def set_availability(self, status):
        self._availability = status

    def __str__(self):
        status = "Available" if self._availability else "Not Available"
        return f"{self._title} by {self._author} | ISBN: {self._isbn} | {status}"
